kymatio/scattering2d/backend/torch_skcuda_backend.py

```diff
@@ -0,0 +1,255 @@
+# Authors: Edouard Oyallon, Sergey Zagoruyko
+
+from collections import namedtuple
+import torch
+from skcuda import cublas
+import cupy
+from string import Template
+from torch.nn import ReflectionPad2d
+
+
+
+BACKEND_NAME = 'torch_skcuda'
+
+@cupy.util.memoize(for_each_device=True)
+def load_kernel(kernel_name, code, **kwargs):
+    code = Template(code).substitute(**kwargs)
+    kernel_code = cupy.cuda.compile_with_cache(code)
+    return kernel_code.get_function(kernel_name)
+
+Stream = namedtuple('Stream', ['ptr'])
+
+def getDtype(t):
+    if isinstance(t, torch.cuda.FloatTensor):
+        return 'float'
+    elif isinstance(t, torch.cuda.DoubleTensor):
+        return 'double'
+
+def iscomplex(x):
+    return x.shape[-1] == 2
+
+def isreal(x):
+    return x.shape[-1] == 1
+
+class SubsampleFourier(object):
+    """Subsampling of a 2D image performed in the Fourier domain.
+
+        Subsampling in the spatial domain amounts to periodization
+        in the Fourier domain, hence the formula.
+
+        Parameters
+        ----------
+        x : tensor_like
+            Input tensor with at least 5 dimensions, the last being the real
+             and imaginary parts.
+            Ideally, the last dimension should be a power of 2 to avoid errors.
+        k : int
+            Integer such that x is subsampled by 2**k along the spatial variables.
+
+        Returns
+        -------
+        res : tensor_like
+            Tensor such that its fourier transform is the Fourier
+            transform of a subsampled version of x, i.e. in
+            FFT^{-1}(res)[u1, u2] = FFT^{-1}(x)[u1 * (2**k), u2 * (2**k)].
+
+    """
+    def __init__(self):
+        self.block = (32, 32, 1)
+
+    def GET_BLOCKS(self, N, threads):
+        return (N + threads - 1) // threads
+
+    def __call__(self, x, k):
+        if not x.is_cuda:
+            raise TypeError('Use the torch backend (without skcuda) for CPU tensors!')
```

Comment:
Can't we do these checks in the front-end, when we input something to the forward function?